### PR TITLE
Fix promo spacing, update home page examples to match frontend

### PR DESCRIPTION
**Pull request by DamienBurkeWN**
### Testing
This is at heart just the reduction of the spacing under the promo banner and before the promo blocks, so verify promo banner and promo block structures display as normal, and that the gap between them on the homepage and memorial page examples is now smaller.

The home page and memorial page examples in Storybook are just that, example layouts - the real layout is decided in the frontend. They'd got out of step so I've made them match, more or less, and added controls so that the number of news stories & promos can be varied, so Richard & Co can see what the spacing looks like in various scenarios.

### CODE REVIEW

File: src/library/pages/Home/HomePage.tsx [hunk 1]
```diff
@@ -3,26 +3,31 @@ import * as PageStructures from '../../structure/PageStructures';
 import AlertBanner from '../../structure/AlertBanner/AlertBanner';
 import HomeUnitarySection from '../../structure/HomeUnitarySection/HomeUnitarySection';
 import { newsArticleData } from '../../structure/NewsArticleFeaturedBlock/NewsArticleFeaturedBlockData';
+import { promoData } from '../../structure/PromoBlock/PromoBlock.storydata';
+import { HeroImageProp } from '../../structure/HomeHero/HomeHero.types';
```
Comment (DamienBurkeWN) [inline src/library/pages/Home/HomePage.tsx:7]: To enable Richard & Co. to view the changed spacing the home page examples needed updating and boy did they need some updating. Lots of copy and pasta of props from elsewhere, which I've now corrected to importing the originals...

File: src/library/pages/Home/HomePage.tsx [hunk 2]
```diff
@@ -3,26 +3,31 @@ import * as PageStructures from '../../structure/PageStructures';
...
+import { PageLinkProp } from '../../structure/ServicesLinksList/ServicesLinksList.types';
 
 export interface HomePageProps {
-  heroArray: Array<HeroImageProp>;
-  servicesArray: Array<PageLinkProp>;
-}
-interface HeroImageProp {
```
Comment (DamienBurkeWN) [inline src/library/pages/Home/HomePage.tsx:11]: The diff is not good here, HomePageProps has been updated to include comments so the Docs page shows them correctly in Storybook, and HeroImageProp is now imported not copied and pasted.

File: src/library/pages/Home/HomePage.tsx [hunk 3]
```diff
@@ -32,8 +37,12 @@ const submitInfo = [{
...
 const AlertMessage = <p>Coronavirus | National lockdown: stay at home. <a href="/">Learn what this means for residents and workers here</a></p>
 
-
-export const HomePage: React.FC<HomePageProps> = ({ heroArray, servicesArray }) => (
+/**
+ * An example home page layout constructed from the structures and components defined in the
+ * design system. The actual page layout is defined within the front end app and may be
+ * out of step with this example.
+ */
+export const HomePage: React.FunctionComponent<HomePageProps> = ({ numberOfPromos = 4, numberOfNewsStories = 3, heroArray, servicesArray }) => (
```
Comment (DamienBurkeWN) [inline src/library/pages/Home/HomePage.tsx:45]: Added ability to alter number of news stories and promo tiles in the example

File: src/library/pages/MemorialPage/MemorialPage.tsx [hunk 4]
```diff
@@ -1,33 +1,30 @@
 import React from 'react';
 import * as PageStructures from '../../structure/PageStructures';
 import AlertBanner from '../../structure/AlertBanner/AlertBanner';
-import HomeUnitarySection from '../../structure/HomeUnitarySection/HomeUnitarySection';
```
Comment (DamienBurkeWN) [inline src/library/pages/MemorialPage/MemorialPage.tsx:4]: Similar treatment in this file plus removal of unused stuff

File: src/library/structure/PromoBanner/PromoBanner.styles.js [hunk 5]
```diff
@@ -2,15 +2,13 @@ import styled from "styled-components";
 
 export const Container = styled.div`
     overflow: hidden;
-    margin: 40px auto;
+    margin: 40px auto 5px auto;
```
Comment (DamienBurkeWN) [inline src/library/structure/PromoBanner/PromoBanner.styles.js:5]: So this is the actual spacing fix...!

File: src/library/pages/Home/HomePage.tsx [hunk 6]
```diff
@@ -98,23 +104,29 @@ export const HomePage: React.FC<HomePageProps> = ({ heroArray, servicesArray })
...
-          ctaText="How to become a volunteer"
-          image1440x810="https://images.unsplash.com/photo-1612277795315-26dd8dcdc8a0?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=2550&q=80"
-          image144x81="https://imgur.com/lIHCSwV.jpg"
-        >
-          <p>We’re looking for people to come forward and help deliver the Covid-19 vaccination programme in North Northamptonshire.</p>
-        </PageStructures.PromoBanner>
+        { numberOfPromos > 0 &&
+          <>
+            <PageStructures.PromoBanner
+              title="Volunteer at a local Covid-19 vaccine centre"
```
Comment (chrisrhymes) [inline src/library/pages/Home/HomePage.tsx:110]: Feels strange having this hard coded for some reason. Is it worth taking the props from a PromoBanner story?

File: src/library/structure/PromoBanner/PromoBanner.styles.js (hunk 5 above)
Comment (chrisrhymes) [reply src/library/structure/PromoBanner/PromoBanner.styles.js:5]: Feels quite cramped to me, but fine if this is the design.

Comment (DamienBurkeWN) [reply src/library/structure/PromoBanner/PromoBanner.styles.js:5]: Yup, approved by Richard now

File: src/library/pages/Home/HomePage.tsx (hunk 6 above)
Comment (DamienBurkeWN) [reply src/library/pages/Home/HomePage.tsx]: Yeah, it was a job half done wasn't it. Well, you've got more to review now ;)

**Review by chrisrhymes — APPROVED** [review]

**Pull request merged**